### PR TITLE
Add mem dump example

### PR DESCRIPTION
**Pull request by Wenzel**
This PR adds a `mem-dump` example to kvmi.
Capture d’écran de 2021-02-19 10-17-11

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -610,6 +650,12 @@ impl KVMIntrospectable for KVMi {
         }
         Ok(max_gfn)
     }
+
+    /// Returns the highest physical address
+    fn get_maximum_paddr(&self) -> Result<u64, KVMiError> {
```
Comment (rageagainsthepc) [inline src/lib.rs:655]: Seems unnecessary and potentially problematic since it assumes that page size is always the same. 🤔

Comment (Wenzel) [reply src/lib.rs:655]: Doing the same thing as libvmi does.
https://github.com/libvmi/libvmi/blob/master/libvmi/driver/xen/xen.c#L792

(btw we might be missing a +1 here)

Comment (rageagainsthepc) [reply src/lib.rs:655]: Yeah, but this crate should only contain safe API bindings, not implement new APIs. I think this function belongs to libmicrovmi.

Comment (rageagainsthepc) [reply src/lib.rs:655]: But, to be fair. We are already doing a lot more in this crate than to provide safe bindings. ¯\\\_(ツ)\_/¯

Comment (Wenzel) [reply src/lib.rs:655]: I implemented this function only to provide a nicer interface when demonstrating the API in `mem-dump.rs` :)
We can keep this small addition I believe :)

**Pull request merged**